src/tas.rs

```diff
@@ -9,33 +10,35 @@ pub struct TasConfig {
     pub base_time: i64,
     pub sched_entries: Vec<String>,
 }
+#[derive(Debug)]
 pub struct TasSchedule {
     pub time: i64,
     pub prio: Vec<i64>,
 }
-pub fn to_ns(input: &Value) -> i64 {
-    if let Some(input) = input.as_str() {
-        let len = input.len();
-        let v = input[..len-2].parse::<i64>().unwrap();
-        let unit = input[len-2..].parse::<String>().unwrap();
+pub fn to_ns(input: &Value) -> Result<i64, String> {
+    if let Some(value) = input.as_str() {
+        let matched = regex::Regex::new(r"^(?P<v>[\d_]+)\s*(?P<unit>|ns|us|µs|ms)$").unwrap().captures(value).unwrap();
+        let v = matched.name("v").unwrap().as_str().parse::<i64>().unwrap();
+        let unit = matched.name("unit").unwrap().as_str();
         return {
-            match unit.as_str() {
-                "ns" => v,
-                "us" => v * 1000,
-                "ms" => v * 1000 * 1000,
-                "s" => v * 1000 * 1000 * 1000,
-                _ => 0,
+            match unit {
+                "" => Ok(v),
+                "ns" => Ok(v),
+                "us" => Ok(v * 1000),
+                "µs" => Ok(v * 1000),
+                "ms" => Ok(v * 1000 * 1000),
+                _ => Err(format!("{} is not valid time", value)),
```

Comment:
Could use `unreachable` here because the regex strictly limits unit into fixed set.
It would be useful if we add more unit to the regex later and forgot to add branch here.